### PR TITLE
Fix nil pointer in extractLiteralMap

### PR DESCRIPTION
**Pull request by mayitbeegh**
Signed-off-by: Sean Lin <sean@union.ai>

# TL;DR
Fix nil pointer dereference in extractLiteralMap. Acceptance tested.
Note: this doesn't fix https://github.com/flyteorg/flyte/issues/1314

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1320

## Follow-up issue
https://github.com/flyteorg/flyte/issues/1314

### CODE REVIEW

**Comment by pmahindrakar-oss** [comment]
@mayitbeegh  i already have this fix in my current PR https://github.com/flyteorg/flytectl/pull/176/files
Can we close this one. Also have added unit tests aswell to cover this scenario
